### PR TITLE
Add secure protocol to server string if it's missing

### LINKED ISSUE #71: Allow entering scheme-less instance URL
## Description

When you enter a plain instance URL without a scheme in the settings screen, login silently fails, but logs an "unsupported URL" error:

```
WriteFreely-MultiPlatform[26399:150343] Task <B894ED7C-6878-4351-AE92-68B928253B19>.<1> finished with error [-1002] Error Domain=NSURLErrorDomain Code=-1002 "unsupported URL" UserInfo={NSLocalizedDescription=unsupported URL, NSErrorFailingURLStringKey=//api/auth/login, NSErrorFailingURLKey=//api/auth/login, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <B894ED7C-6878-4351-AE92-68B928253B19>.<1>"
), _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <B894ED7C-6878-4351-AE92-68B928253B19>.<1>, NSUnderlyingError=0x600002f60ae0 {Error Domain=kCFErrorDomainCFNetwork Code=-1002 "(null)"}}
```

## Steps to reproduce

1. Go to Settings screen
1. Log out, if necessary
1. Enter username: `demo`, password: `demo`, and instance URL: `pencil.writefree.ly`
1. Notice login failure and error log

## Expected Result

I'm logged into `https://pencil.writefree.ly`

### Design

From a product standpoint, I think we can assume and automatically prepend `https://` if no URL scheme is supplied. If that fails, we should show an error message to the user, e.g. "Please enter a full valid URL, starting with https://"

**Note for the future**: We don't need to do this now, but eventually we _should_ retry with `http://` if the automatically-prepended `https://` URL fails. And then show a relevant error message if that also fails.

## Actual result

Login form was reset to its original state with no indication of what happened.

### PR DESCRIPTION
**Pull request by AngeloStavrow**
Closes #71.

This PR adds a check in the AccountLoginView — when the user taps the login button, it prepends the "https://" prefix to the server string if it's missing before logging in.

### CODE REVIEW

File: Shared/Account/AccountLoginView.swift [hunk 1]
```diff
@@ -52,6 +52,10 @@ struct AccountLoginView: View {
                     .padding()
             } else {
                 Button(action: {
+                    let secureProtocolPrefix = "https://"
+                    if !server.hasPrefix(secureProtocolPrefix) {
```
Comment (thebaer) [inline Shared/Account/AccountLoginView.swift:56]: I think we should check for `http://` too, and only prepend something if neither `http` nor `https` is given. I don't imagine many people will be running insecure instances, but I could imagine situations like testing on a local network, where you're accessing an instance at `http://10.0.0.10`.

**Review by thebaer — CHANGES_REQUESTED** [review]

File: Shared/Account/AccountLoginView.swift (hunk 1 above)
Comment (AngeloStavrow) [reply Shared/Account/AccountLoginView.swift:56]: Yes! Good catch — I'm making changes now.

Comment (AngeloStavrow) [reply Shared/Account/AccountLoginView.swift:56]: @thebaer Given that [insecure connections are blocked by default](https://github.com/writeas/writefreely-swiftui-multiplatform/issues/71#issuecomment-698927571), any attempt to login over http will fail. I propose that for the 1.0 release, if an http connection is attempted, we should try to upgrade to https (and put up an alert if it fails). What do you think?

Comment (AngeloStavrow) [reply Shared/Account/AccountLoginView.swift]: 👇 Added those changes but will leave this conversation for you to resolve.

Comment (thebaer) [reply Shared/Account/AccountLoginView.swift]: Okay, no problem. The current solution looks good for now!

**Review by thebaer — APPROVED** [review]

**Pull request merged**